File: lib/src/animate/css_animation_options.dart
library angular2.src.animate.css_animation_options;

class CssAnimationOptions {
  /** initial styles for the element */
  Map<String, dynamic> fromStyles;
  /** destination styles for the element */
  Map<String, dynamic> toStyles;
  /** classes to be added to the element */
  List<String> classesToAdd = [];
  /** classes to be removed from the element */
  List<String> classesToRemove = [];
  /** classes to be added for the duration of the animation */
  List<String> animationClasses = [];
  /** override the duration of the animation (in milliseconds) */
  num duration;
  /** override the transition delay (in milliseconds) */
  num delay;
}

File: ts/src/animate/css_animation_options.ts
export class CssAnimationOptions {
  /** initial styles for the element */
  fromStyles: {[key: string]: any};

  /** destination styles for the element */
  toStyles: {[key: string]: any};

  /** classes to be added to the element */
  classesToAdd: string[] = [];

  /** classes to be removed from the element */
  classesToRemove: string[] = [];

  /** classes to be added for the duration of the animation */
  animationClasses: string[] = [];

  /** override the duration of the animation (in milliseconds) */
  duration: number;

  /** override the transition delay (in milliseconds) */
  delay: number;
}
